src/rapids_dependency_file_generator/_rapids_dependency_file_generator.py

```diff
@@ -438,18 +483,50 @@ def make_dependency_files(
                 )
                 contents = make_dependency_file(
                     file_type=file_type,
-                    name=full_file_name,
+                    conda_env_name=os.path.splitext(full_file_name)[0],
+                    file_name=full_file_name,
                     config_file=parsed_config.path,
                     output_dir=output_dir,
-                    conda_channels=prepend_channels + parsed_config.channels,
+                    conda_channels=conda_channels,
                     dependencies=deduped_deps,
                     extras=file_config.extras,
                 )
 
                 if to_stdout:
-                    print(contents)
+                    if len(file_keys) == 1:
+                        print(contents)
+                    else:
+                        all_dependencies.update(deduped_deps)
                 else:
                     os.makedirs(output_dir, exist_ok=True)
                     file_path = os.path.join(output_dir, full_file_name)
                     with open(file_path, "w") as f:
                         f.write(contents)
+
+    # create one unified output from all the file_keys, and print it to stdout
+    if to_stdout and len(file_keys) > 1:
+        # convince mypy that 'output' is not None here
+        #
+        # 'output' is technically a set because of https://github.com/rapidsai/dependency-file-generator/pull/74,
+        # but since https://github.com/rapidsai/dependency-file-generator/pull/79 it's only ever one of the following:
+        #
+        #   - an exactly-1-item set (stdout=True, or when used by rapids-build-backend)
+        #   - 'None' (stdout=False)
+        #
+        err_msg = (
+            "Exactly 1 output type should be provided when asking rapids-dependency-file-generator to write to stdout. "
+            "If you see this, you've found a bug. Please report it at https://github.com/rapidsai/dependency-file-generator/issues."
+        )
+        assert output is not None, err_msg
+
+        contents = make_dependency_file(
+            file_type=output.pop(),
+            conda_env_name="rapids-dfg-combined",
+            file_name="ignored-because-multiple-pyproject-files-are-not-supported",
```

Comment:
`make_dependency_file()` is already really 3 distinct codepaths inside one function. The codepath for generating `pyproject` output rightly expects to be given a path to an existing `pyproject.toml` file to read in.

I'm proposing just not allowing multiple `--file-key` together with `--output pyproject` and passing through this hard-coded nonsense string instead of taking on the complexity of other options like:

* breaking `make_dependency_file()` up into e.g. `make_conda_env_file()`, `make_requirements_file()`, and `make_pyproject_file()`
* allowing `file_name=None` to be passed to `make_dependency_file()` and generating TOML content from scratch if it is

Reply:
Let me try a different way with links.

After processing `dependencies.yaml`, filtering stuff by output type and matrices, deduplicating, etc., DFG wants the literal text that it'll either echo to `stdout` or write to a file.

For that, it calls `make_dependency_file()`:

https://github.com/rapidsai/dependency-file-generator/blob/6e1081109351711c2c81c271861e0b94eb305edd/src/rapids_dependency_file_generator/_rapids_dependency_file_generator.py#L439

Despite the name, that function actually just creates and returns that text... it never writes anything to the filesystem.

The body of the function, in pseudocode, is like:

```python
def make_dependency_file(...):
    relative_path_to_config_file = os.path.relpath(config_file, output_dir)
    file_contents = textwrap.dedent(
        f"""\
        {HEADER}
        # To make changes, edit {relative_path_to_config_file} and run `{cli_name}`.
        """
    )
    if conda:
        # {code specific to conda env YAML files}
    elif requirements:
        # {code specific to requirements.txt}
    elif pyproject:
        # {code specific to pyproject.toml}

    return file_contents
```

So there's very little shared code in there. The list of arguments for the function contains a mix of things that are only used for some but not all of the output types.

Like this:

https://github.com/rapidsai/dependency-file-generator/blob/6e1081109351711c2c81c271861e0b94eb305edd/src/rapids_dependency_file_generator/_rapids_dependency_file_generator.py#L117-L119

That's complex but has been kind of "fine" as long as this was always being called with the data from a single `--file-key`.

For this PR, we now want to use that function to generate a type out of output from multiple `--file-key` entries. For that, there's no single representative `output_dir`, for example. Here, I've just kind of awkwardly provided values with the right types so `mypy` will be happy, with the knowledge that they don't really matter (since most of the arguments are `pyproject`-specific, and this PR isn't supporting pyproject.toml).

It'd be clearer if `make_dependency_file()` was decomposed into `make_conda_env_file()`, `make_requirements_file()` and `make_pyproject_file()`. But I really really did not want to take that on in the scope of this PR.